src/main/java/edu/ucsb/cs156/rec/entities/RecommendationRequest.java

```diff
@@ -0,0 +1,38 @@
+package edu.ucsb.cs156.rec.entities;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.time.LocalDateTime;
+
+/**
+ * This is a JPA entity that represents a RecommendationRequest, i.e. an entry
+ * that comes from the RecommendationRequest API for requests from students.
+ */
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@Entity(name = "recommendationrequest")
+public class RecommendationRequest {
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private long id;
+  private String status; 
+
+  private String professorName;
+  private String professorEmail;
+  private String requesterName;
```

Comment:
Name isn't unique and gives us no way to ensure that the user that creates this request can be tied back to it.  We want to be able to show the user *only their* requests, and this database design gives us no way to do that.

Replace with: `private Long requesterId` or better yet:

```java
  @ManyToOne
  @JoinColumn(name = "requester_id")
  private User requester;
```  

This will give you access in a RecommendationRequest object `r` to:
```java
  r. getRequester().getName();
  r.getRequester().getEmail();
 // etc...
```

See: https://ucsb-cs156.github.io/topics/spring_boot/spring_boot_database.html#special-case-records-owned-by-a-user-an-intro-to-foreign-keys for more details

Reply:
Fixed